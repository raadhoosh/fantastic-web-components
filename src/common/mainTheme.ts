import { Theme } from './styled';
import * as Color from 'color';
function darken(hex: string) {
    return Color(hex)
        .darken(0.5)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#0066b1',
        secondary_dark: darken('#0066b1'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    bottom: {
        fontsize: '1rem',
        border: '5px',
    },
    darken: (_color?: string) => {
        return Color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return Color(hex).fade(a ? a : 0.5).string();
    },
};

export default mainTheme;
