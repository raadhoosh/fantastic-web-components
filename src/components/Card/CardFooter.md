CardFooter example:

```js
<CardFooter><Button>Accept</Button> </CardFooter>
```