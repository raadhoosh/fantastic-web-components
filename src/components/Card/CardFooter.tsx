import styled from '../../common/styled';
import * as React from 'react';
export interface IProps {
    /** This is a primary button it out!. */
    primary?: boolean;
    /** This is a secondary button it out!. */
    secondary?: boolean;
    /** This is a success button it out!. */
    success?: boolean;
    /** This is a info button it out!. */
    info?: boolean;
    /** This is a warning button it out!. */
    warning?: boolean;
    /** This is a danger button it out!. */
    danger?: boolean;
    /** This is a light button it out!. */
    light?: boolean;
    /** This is a dark button it out!. */
    dark?: boolean;
    /** This is a text color button it out! */
    children?: any;
    /** This is a Background Color button it out!. */
    backgroundColor?: string;
    /** This is a Text Color button it out!. */
    color?: string;
    /** This is a margin button it out!. */
    margin?: string;
    radius?: string;
    bgColor?: string;
    borderColor?: string;
    /** This is a noShadow button it out!. */
    noShadow?: boolean;
    theme?: any;
    size?: 'sm' | 'lg' | 'md';
}
const EmCardFooter = styled('div')(
    (props: (IProps)) => {
        const base = {
            borderTop: '1px solid',
            padding: '.75rem .75rem',
            marginTop: 0,
            backgroundColor: 'transparent',
            borderRadius: 'none',
            borderTopColor: 'transparent',
            ':first-child': {
                borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) 0 0'
            }
        };
        if (props.radius) {
            base.borderRadius = props.radius;
        }
        if (props.bgColor) {
            if (props.theme.color[props.bgColor]) {
                base.backgroundColor = props.theme.color[props.bgColor];
            } else {
                base.backgroundColor = props.bgColor;
            }
        }
        const color = (props.primary && 'primary') ||
            (props.secondary && 'secondary') ||
            (props.success && 'success') ||
            (props.info && 'info') ||
            (props.danger && 'danger') ||
            (props.warning && 'warning') ||
            (props.light && 'light') ||
            (props.dark && 'dark') ||
            ('');
        if (color !== '') {
            base.backgroundColor = props.theme.color[color];
        }
        if (props.borderColor) {
            if (props.theme.color[props.borderColor]) {
                base.borderTopColor = props.theme.color[props.borderColor];
            } else {
                base.borderTopColor = props.borderColor;
            }
        }
        return base;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const CardFooter = (props: IProps) => {

    return (
        <EmCardFooter {...props}> {props.children} </EmCardFooter>
    );
};
export default CardFooter;
