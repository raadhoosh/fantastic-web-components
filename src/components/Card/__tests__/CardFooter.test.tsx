import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import CardFooter from '../CardFooter';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const tree = renderer.create(<CardFooter theme={mainTheme} primary>hi test</CardFooter>).toJSON();
    // expect(tree).toHaveStyleRule('background-color', mainTheme.color.primary);
    // expect(tree).toHaveStyleRule('color', mainTheme.color.primary_dark);
    expect(tree).toMatchSnapshot();
});
