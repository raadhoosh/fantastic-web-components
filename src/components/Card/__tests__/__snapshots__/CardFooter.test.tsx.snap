// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<div
  className="css-1r7od4f"
>
   
  hi test
   
</div>
`;
